### PR TITLE
[docs] Add grid based layout component

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Alternative CSS Grid approach to #16206

This adds the basic layout component that the following layout.

```
     (width: 100%)
|--------- A ---------| (height: 60px)
|- B -|--- C ---|- D -| (height: 100vh - 60px)
```

- **A** — Requires banner header which is always visible
    - → Always show
- **B** — Optional navigation sidebar on the left
    - → Hide when empty, or _window_ is `<=768px` (custom) 
- **C** — The page content (`width: 100%; height: 100%`)
    - → Always show
    - → Set max width of `1200px` (`breakpoints.large`), align center when space is available
    - → Add custom scroll bars when height overflows
- **D** — Optional table of contents sidebar on the right (`width: min-content; height: 100%`)
    - → Hide when empty, or _window_ is `<=900px` (`breakpoints.medium`)

## Screenshots

<details><summary>Used code to generate screenshots below</summary>

```jsx
import { css } from '@emotion/react';
import React from 'react';

import { Layout } from '~/ui/components/Layout';

export default function TestPage() {
  const layoutProps = {
    header: <div css={headerStyle}>A</div>,
    navigation: <div css={navigationStyle}>B</div>,
    sidebar: <div css={sidebarStyle}>D</div>,
  };

  return (
    <Layout {...layoutProps}>
      <div css={contentStyle}>{fill(100)}</div>
    </Layout>
  );
}

function fill(length: number) {
  return Array(length).fill(<>C<br /></>);
}

const block = {
  display: 'flex',
  width: '100%',
  minHeight: '100%',
  justifyContent: 'center',
  alignItems: 'center',
  color: 'black',
  fontWeight: 'bold',
};

const headerStyle = css({
  ...block,
  backgroundColor: 'yellow',
});

const navigationStyle = css({
  ...block,
  backgroundColor: 'red',
  width: '200px',
});

const contentStyle = css({
  ...block,
  backgroundColor: 'green',
});

const sidebarStyle = css({
  ...block,
  backgroundColor: 'blue',
  width: '200px',
});
```

</details>

<details><summary>1. normal screen width with all columns</summary>

localhost_3002_test_ (2)

</details>

<details><summary>2. normal screen width without <b>D</b> content</summary>

localhost_3002_test_ (3)

</details>

<details><summary>3. normal screen width without <b>B</b> and <b>D</b> content</summary>

localhost_3002_test_ (8)

</details>

<details><summary>4. large screen width with all columns</summary>

localhost_3002_test_ (4)

</details>

<details><summary>5. small screen width auto-hiding <b>D</b></summary>

localhost_3002_test_ (5)

</details>

<details><summary>6. tiny screen width auto-hiding <b>B</b> and <b>D</b></summary>

localhost_3002_test_ (6)

</details>

<details><summary>7. normal screen width auto-scrolling <b>C</b></summary>

image

</details>

# How

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/ui/components/Layout/Layout.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      <div css={sidebarStyle}>{sidebar}</div>
+    </div>
+  );
+}
+
+const layoutStyle = css({
+  display: 'grid',
+  height: '100vh',
+  gridTemplateRows: '60px calc(100vh - 60px)',
+  gridTemplateColumns: 'min-content auto min-content',
```
Comment (jonsamp) [inline docs/ui/components/Layout/Layout.tsx:35]:
```suggestion
  gridTemplateColumns: '280px auto 280px',
```

We may want to make this even more forceful to ensure that child pages cannot change the size of `min-content`.

**Review by jonsamp — APPROVED** [review]
I think this approach is an improvement and will be nicer to maintain.

File: docs/ui/components/Layout/Layout.tsx (hunk 1 above)
Comment (byCedric) [reply docs/ui/components/Layout/Layout.tsx:35]: I had problems with the responsiveness of the grid with these values. When we don't render the right sidebar at all, we probably still want to use that space. That doesn't work with these fixed values, and `fit-content(280px)` does not actually limits the column's width.

Comment (byCedric) [reply docs/ui/components/Layout/Layout.tsx:35]: For now, I'll merge this without the suggestion, we can circle back to it later. I want to start composing the new designs with sidebars to prepare for tweaking and polishing.

**Pull request merged**